src/database/operations/generators/generateLoadItemOperation.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Generates an item loader function for a specific query and table.
+ *
+ * @template T - The type of the item being loaded.
+ * @template P - The type of the parameters for the query. Use labeled tuples so the generated function has a pretty type definition.
+ *
+ * @param {string} queryName - The name of the query to execute.
+ * @param {string} entityType - The name of entity type being loaded, for use in error messages.
+ * @param {Object} parameterNames - An object mapping parameter indices to their names within the query.
+ *
+ * @returns {Function} A function that takes query parameters, limit, and offset, and returns a promise that resolves to a bundle of entries and total count.
+ */
```

Comment:
This documentation is excellent and I want to see it when I poke at the calls to this function in other parts of the code. But codium doesn't show it unless it precedes the function, a few lines down.